enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/upstream/strategies/TypicallyConnectToRandomReadReplicaStrategy.java

```diff
@@ -54,18 +62,15 @@ public TypicallyConnectToRandomReadReplicaStrategy( int connectToCoreInterval )
         }
         else
         {
-            Optional<MemberId> memberId = getReadReplicaMemberId();
-            if ( !memberId.isPresent() )
-            {
-                memberId = getCoreMemberId();
-            }
-            return memberId;
-        }
-    }
+            // shuffled members
+            List<MemberId> readReplicaMembers = new ArrayList<>( topologyService.localReadReplicas().members().keySet() );
+            Collections.shuffle( readReplicaMembers );
 
-    private Optional<MemberId> getReadReplicaMemberId()
-    {
-        return topologyService.localReadReplicas().randomReadReplicaMemberId();
```

Comment:
I think this is the only call to randomReadReplicaMemberId so the method can be removed

Reply:
Yes, it's been removed

Reply:
Not very important, but I can't see that there is any change to ReadReplicatTopology